docs/release-notes/trainer-api.rst

```diff
@@ -0,0 +1,10 @@
+:orphan:
```

Comment:
Glad this has the breaking changes, but it also needs a feature announcement.  This is a pretty big-deal feature.